### ISSUE TITLE
✨ New `orNull` and `orThrow` functions for `EmailAddress`

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#669]
## 📝 Description

For aligning with the Application Programming Interface (API) design of the `Zero` type (#668), we want to introduce the following **experimental** functions:

```kotlin
fun EmailAddress.Companion.orNull(text: Any): EmailAddress?
fun EmailAddress.Companion.orNull(text: Any, pattern: Any): EmailAddress?

fun EmailAddress.Companion.orThrow(text: Any): EmailAddress
fun EmailAddress.Companion.orThrow(text: Any, pattern: Any): EmailAddress
```

These functions should replace the `fromString` and the `fromStringOrNull` **experimental** functions.

## ✅ Checklist

- [x] ✨ Add the `orNull(Any)` function with tests, documentation and samples.
- [x] ✨ Add the `orNull(Any, Any)` function with tests, documentation and samples.
- [x] ✨ Add the `orThrow(Any)` function with tests, documentation and samples.
- [x] ✨ Add the `orThrow(Any, Any)` function with tests, documentation and samples.
- [ ] 📝 Add references to `orNull` and `orThrow` functions in their documentation.
- [ ] 🔥 Remove the `fromStringOrNull(Any)` function for using the `orNull(Any)` one instead.
- [ ] 🔥 Remove the `fromStringOrNull(Any, Any)` function for using the `orNull(Any, Any)` one instead.
- [ ] 🔥 Remove the `fromString(Any)` function for using the `orThrow(Any)` one instead.
- [ ] 🔥 Remove the `fromString(Any, Any)` function for using the `orThrow(Any, Any)` one instead.
- [ ] ♻️ Refactor the `orNull` and the `orThrow` functions.
- [ ] 📝 Update the unreleased changelog for this issue.

**Comment by LVMVRQUXL** [comment]
Superseded by #671.

**Comment by LVMVRQUXL** [comment]
These functions will be implemented instead of those suggested by #675.

**Comment by LVMVRQUXL** [comment]
Superseded by #677.
